### PR TITLE
TEST-#24: Add tests run to CI

### LINKED ISSUE #24: Add tests run to CI
This is a subtask of #9.

### PR DESCRIPTION
**Pull request by prutskov**
Signed-off-by: Alexey Prutskov <alexey.prutskov@intel.com>

## What do these changes do?

- [x] passes `flake8 .`
- [x] passes `black .`
- [x] signed commit with `git commit -s` 
- [x] Resolves #24  
- [x] tests added and passing
- [ ] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -47,3 +47,34 @@ jobs:
           architecture: "x64"
       - run: pip install flake8 flake8-print
       - run: flake8 --enable=T .
+
+  test-all:
```
Comment (YarShev) [inline .github/workflows/ci.yml:51]: test-all should include Windows run too.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -47,3 +47,73 @@ jobs:
...
+          environment-file: environment.yml
+          python-version: ${{matrix.python-version}}
+          channel-priority: strict
+          use-only-tar-bz2: true # IMPORTANT: This needs to be set for caching to work properly
+          auto-update-conda: true # this enable `use-only-tar-bz2` feature on Windows
+          # Miniconda setup sometimes fails because of an http error. retry
+          # it once if it fails. todo(https://github.com/conda-incubator/setup-miniconda/issues/129):
+          # get rid of this once setup-miniconda can be configured to retry
+          # setup on http errors.
+          max_attempts: 2
```
Comment (YarShev) [inline .github/workflows/ci.yml:111]: Do we have frequent failures too?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -47,3 +47,73 @@ jobs:
...
+        with:
+          activate-environment: unidist
+          environment-file: environment.yml
+          python-version: ${{matrix.python-version}}
+          channel-priority: strict
+      - name: Conda environment
+        run: |
+          conda info
+          conda list
+      - run: python unidist/cli -m pytest unidist/test/test_general.py
```
Comment (YarShev) [inline .github/workflows/ci.yml:78]: Do we have a freeze of tests for MP backend if we run those as unidist/test on linux?

File: unidist/test/test_general.py [hunk 4]
```diff
@@ -2,28 +2,44 @@
...
+    elif source == "task":
+        object_ref = task.remote(3)
+    else:
+        object_ref = TestActor.remote().task.remote(1)
     assert_equal(unidist.is_object_ref(object_ref), True)
 
 
+@pytest.mark.skipif(
+    Backend.get() == BackendName.MP,
+    reason="Hangs on `multiprocessing` backend. Details are in https://github.com/modin-project/unidist/issues/64.",
```
Comment (YarShev) [inline unidist/test/test_general.py:41]: Does this happen if we run this test separately?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (prutskov) [reply .github/workflows/ci.yml:111]: No, this is a proactive fix for possible failures due to connection problems on Windows

File: .github/workflows/ci.yml (hunk 3 above)
Comment (prutskov) [reply .github/workflows/ci.yml]: No. Using tests changes from this PR (skipif for some MP cases) we can run all test scope at once. Fixed

File: unidist/test/test_general.py (hunk 4 above)
Comment (prutskov) [reply unidist/test/test_general.py:41]: For small amount of workers (2 as used in CI) this happens for separate runs/all scope.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (YarShev) [reply .github/workflows/ci.yml:109]: We see such an error on linux in Modin too. Should we add restart for linux?

File: .github/workflows/ci.yml (hunk 3 above)
Comment (YarShev) [reply .github/workflows/ci.yml]: Ok, let's leave running the whole test scope.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (prutskov) [reply .github/workflows/ci.yml:116]: Since we added this for WIndows, for Linux it would be useful too. Added

File: unidist/test/test_actor.py [hunk 5]
```diff
@@ -2,33 +2,56 @@
...
+    reason="Details are in https://github.com/modin-project/unidist/issues/70.",
+)
 def test_num_returns():
     actor = TestActor.remote(7)
     object_ref0, object_ref1 = actor.multiple_returns.options(num_returns=2).remote(3)
     assert_equal(object_ref0, 3)
     assert_equal(object_ref1, 7)
 
 
+@pytest.mark.skipif(
```
Comment (YarShev) [inline unidist/test/test_actor.py:47]: We should be doing this as P0 to get undist fully working on Linux.

Comment (YarShev) [reply unidist/test/test_actor.py:47]: of course, as part of a separate issue

File: unidist/test/test_general.py [hunk 6]
```diff
@@ -2,28 +2,44 @@
...
 from .utils import assert_equal, task, TestActor
 
 unidist.init()
 
 
-# FIXME: If we run all tests (i.e. pytest unidist/test), the tests are hanging.
-# However, if we remove third argument, the tests will be passed. This is true for MultiProcessing backend and
-# requires an investigation.
+@pytest.mark.skipif(
+    Backend.get() == BackendName.MP,
```
Comment (YarShev) [inline unidist/test/test_general.py:18]: see the comment above

File: unidist/test/test_general.py [hunk 7]
```diff
@@ -2,28 +2,44 @@
...
+    if source == "put":
+        object_ref = unidist.put(1)
+    elif source == "task":
+        object_ref = task.remote(3)
+    else:
+        object_ref = TestActor.remote().task.remote(1)
     assert_equal(unidist.is_object_ref(object_ref), True)
 
 
+@pytest.mark.skipif(
```
Comment (YarShev) [inline unidist/test/test_general.py:39]: see the comment above

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -47,3 +47,76 @@ jobs:
           architecture: "x64"
       - run: pip install flake8 flake8-print
       - run: flake8 --enable=T .
+
```
Comment (YarShev) [inline .github/workflows/ci.yml:50]: Should we get a coverage report?

Comment (prutskov) [reply .github/workflows/ci.yml:50]: I don't see the point yet, since we don't have a goal for coverage

File: unidist/test/test_actor.py (hunk 5 above)
Comment (prutskov) [reply unidist/test/test_actor.py:47]: As mentioned, will be fixed in #65

File: unidist/test/test_general.py (hunk 6 above)
Comment (prutskov) [reply unidist/test/test_general.py:18]: https://github.com/modin-project/unidist/pull/56#discussion_r766485355

File: unidist/test/test_general.py (hunk 7 above)
Comment (prutskov) [reply unidist/test/test_general.py:39]: https://github.com/modin-project/unidist/pull/56#discussion_r766485355

File: .github/workflows/ci.yml (hunk 8 above)
Comment (YarShev) [reply .github/workflows/ci.yml:50]: We would see the places which we are not covering and probably are hiding possible errors/bugs.

Comment (prutskov) [reply .github/workflows/ci.yml:50]: We can try to add this as a part of #9 epic, not in this PR.

Comment (YarShev) [reply .github/workflows/ci.yml:50]: Okay, then please issue a PR related to coverage report to see how it goes.

**Review by YarShev — APPROVED** [review]

**Pull request merged**